mgsr/app.js

```diff
@@ -0,0 +1,398 @@
+  }
+  return 0;
+});
+userIds.push(["tag this person directly in Discord", "(Unlisted User)"]);
+
+const list = document.querySelectorAll("[name='playerlist']");
+
+userIds.forEach((item) => {
+  let option = document.createElement("option");
+  option.value = item[1];
+  option.textContent = item[1];
+  option.setAttribute("data-userid", item[0]);
+  let option2 = option.cloneNode(true);
+  let option3 = option.cloneNode(true);
+  let option4 = option.cloneNode(true);
+  list[0].appendChild(option);
+  list[1].appendChild(option2);
+  list[2].appendChild(option3);
+  list[3].appendChild(option4);
+});
+
+function updateState() {
+  // Update global object with selected values
+  // let firstPlace = document.getElementById("placement1");
+  let places = [
+    document.getElementById("placement1"),
+    document.getElementById("placement2"),
+    document.getElementById("placement3"),
+    document.getElementById("placement4"),
+  ];
+  let userIDs = [
+    places[0].nextElementSibling.selectedOptions[0].getAttribute("data-userid"),
+    places[1].nextElementSibling.selectedOptions[0].getAttribute("data-userid"),
+    places[2].nextElementSibling.selectedOptions[0].getAttribute("data-userid"),
+    places[3].nextElementSibling.selectedOptions[0].getAttribute("data-userid"),
+  ];
+  let userNames = [
+    places[0].nextElementSibling.selectedOptions[0].value,
+    places[1].nextElementSibling.selectedOptions[0].value,
+    places[2].nextElementSibling.selectedOptions[0].value,
+    places[3].nextElementSibling.selectedOptions[0].value,
+  ];
+  Object.assign(gameState, {
+    1: {
+      place: places[0].value,
+      userID: userIDs[0],
+      userName: userNames[0],
+    },
+    2: {
+      place: places[1].value,
+      userID: userIDs[1],
+      userName: userNames[1],
+    },
+    3: {
+      place: places[2].value,
+      userID: userIDs[2],
+      userName: userNames[2],
+    },
+    4: {
+      place: places[3].value,
+      userID: userIDs[3],
+      userName: userNames[3],
+    },
+  });
+
+  renderCommands(gameState);
+}
+
+function renderCommands(state) {
+  // Need to output the individual commands for each pairing
+  // For a 1,2,3,4 match, this would be 6 commands
+  // 1: 1v2, 1v3, 1v4; 2: 2v3, 2v4; 3: 3v4
+  // For a 1,2,3 (or 1,1,3), it would be 3 commands
+  // for a 1,2 (or 1,1), just 1 command
+
+  // First, clear any existing elements produced on a previous click
+  document.getElementById("MultiResultsArea").innerHTML = "";
+
+  let pairings = getPairings(state);
+  let hasEnoughSelections = false;
+  let pairList = pairings.map((e) => {
+    const p1 = e[0];
+    const p2 = e[1];
+    if (p1.place != '' &&  p1.userID != null && p2.place != '' && p2.userID != null) {
+      // Only include results in output if it is for a valid pairing
+      // Which means, the pairing has a place and user selected
+
+      const commandID = p1.userID + "_" + p2.userID;
+
+      let p1BotPlace = "1";
+      let p2BotPlace = "2";
```

Comment:
Will always use "#1" and "#2" in the bot command, even if it's the 2nd place player beating the 4th place player, like how people input it today